### ISSUE TITLE
[Crash]: 

### ISSUE THREAD

**Issue by Storozhevoy** [OpenTTD/OpenTTD#11167]
### Version of OpenTTD

OpenTTD 13.3.0.45523 windows 10.0.22621 build 22621

### Steps to reproduce

Ship # 12 is lost and the game slowed down and crashed.

### Upload crash files

[crash20230803112029 (2).zip](https://github.com/OpenTTD/OpenTTD/files/12250248/crash20230803112029.2.zip)
[crash20230803112029.zip](https://github.com/OpenTTD/OpenTTD/files/12250249/crash20230803112029.zip)

**Comment by glx22** [comment]
```
>	[Cadre en ligne] openttd.exe!HasBit(const RailTypes) Ligne 105	C++
 	openttd.exe!HasRailtypeAvail(const Owner company, const RailType railtype) Ligne 188	C++
 	openttd.exe!ValParamRailtype(const RailType rail) Ligne 208	C++
 	openttd.exe!ShowBuildRailToolbar(RailType) Ligne 860	C++
 	openttd.exe!MenuClickBuildRail(int index) Ligne 924	C++
 	openttd.exe!MainToolbarWindow::OnDropdownSelect(int widget, int index) Ligne 2040	C++
 	openttd.exe!Window::OnDropdownClose(Point pt, int widget, int index, bool instant_close) Ligne 305	C++
 	openttd.exe!DropdownWindow::Close() Ligne 200	C++
 	openttd.exe!SetFocusedWindow(Window * w) Ligne 472	C++
 	openttd.exe!Window::InitializeData(int window_number) Ligne 1444	C++
 	openttd.exe!Window::FinishInitNested(int window_number) Ligne 1794	C++
 	openttd.exe!NewsWindow::NewsWindow(WindowDesc * desc, const NewsItem * ni) Ligne 311	C++
 	openttd.exe!ShowNewspaper(const NewsItem * ni) Ligne 621	C++
 	openttd.exe!MoveToNextNewsItem() Ligne 741	C++
 	openttd.exe!DeleteNewsItem(NewsItem * ni) Ligne 778	C++
 	openttd.exe!DeleteVehicleNews(unsigned int vid, unsigned int news) Ligne 929	C++
 	openttd.exe!ChooseShipTrack(Ship * v, TileIndex tile, DiagDirection enterdir, TrackBits tracks) Ligne 498	C++
 	openttd.exe!ShipController(Ship * v) Ligne 776	C++
 	openttd.exe!Ship::Tick() Ligne 846	C++
 	openttd.exe!CallVehicleTicks() Ligne 973	C++
 	openttd.exe!StateGameLoop() Ligne 1415	C++
 	openttd.exe!GameLoop() Ligne 1513	C++
 	openttd.exe!VideoDriver::GameLoop() Ligne 38	C++
 	openttd.exe!VideoDriver::GameThread() Ligne 47	C++
 	openttd.exe!StartNewThread::__l3::<lambda>(const char * name, void(*)(VideoDriver *) && F, VideoDriver * && <A_0>) Ligne 65	C++
 	[Code externe]	
 	openttd.exe!thread_start<unsigned int (__cdecl*)(void *),1>(void * const parameter) Ligne 97	C++
 	[Code externe]	
```

Duplicate of #11043

**Comment by LordAro** [comment]
(fixed in 13.4, please upgrade)
